src/components/form/form.types.ts

```diff
@@ -154,21 +154,54 @@ export interface FormComponentOptions {
     props?: Record<string, any>;
 }
 
-export interface FormLayoutOptions {
+export interface FormLayoutOptions<T = FormLayoutType.Default> {
     /**
      * The type of layout to use
      */
-    type: FormLayoutType;
+    type: T;
 
     /**
-     * When specified on a component within a layout, the component will take
-     * up the full width of the form
+     * @deprecated use `colSpan` instead
      */
     span?: 'all';
 }
 
+export interface GridLayoutOptions
+    extends FormLayoutOptions<FormLayoutType.Grid> {
+    /**
+     * When specified on a component within the grid, the component will take
+     * up the the specified number of columns in the form
+     */
+    // eslint-disable-next-line no-magic-numbers
+    colSpan?: 1 | 2 | 3 | 4 | 5 | 'all';
+
+    /**
+     * When specified on a component within the grid, the component will take
+     * up the the specified number of rows in the form
+     */
+    rowSpan?: number;
+
+    /**
+     * Number of columns to use in the layout
+     */
+    // eslint-disable-next-line no-magic-numbers
+    columns?: 1 | 2 | 3 | 4 | 5;
+
+    /**
+     * Attempts to fill in holes earlier in the grid, if smaller items come up
+     * later. This may cause items to appear out-of-order, when doing so would
+     * fill in holes left by larger items. Defaults to `true`
+     */
+    dense: boolean;
```

Comment:
@jgroth Can't this also be optional?

Reply:
Yes! It should be! I'm not sure why it's not, must be a mistake :sweat_smile: